src/amqp.js

```diff
@@ -0,0 +1,340 @@
+    try {
+      errorUndefinedQueue(queue);
+      const channelInfo = await channel.assertQueue(queue, {durable: true});
+      return channelInfo.messageCount;
+    } catch (error) {
+      handleAmqpErrors(error);
+    }
+  }
+
+  // MARK: Get chunk
+  /**
+   * Gets 100 messages from queue. CHUNK_SIZE set in constants.
+   * Get chunk uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @returns {object|boolean} returns {headers, records, messages} or false
+   */
+  async function getChunk({blobId, status}) {
+    const queue = generateQueueId({blobId, status});
+    try {
+      errorUndefinedQueue(queue);
+      const channelInfo = await channel.assertQueue(queue, {durable: true});
+      if (checkMessageCount(channelInfo)) {
+        debug(`Prepared to getChunk from queue: ${queue}`);
+        // getData: next chunk (100) messages
+        const messages = await getData(queue);
+
+        // Note: headers are from the first message of the chunk
+        debug(`getChunk (${messages ? messages.length : '0'} from queue ${queue}) to records`);
+
+        const headers = getHeaderInfo(messages[0]);
+        const records = await messagesToRecords(messages);
+        return {headers, records, messages};
+      }
+
+      return false;
+    } catch (error) {
+      handleAmqpErrors(error);
+    }
+  }
+
+  // MARK: Get one
+  /**
+   * Gets 1 messages from queue.
+   * Get one uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @returns {object|boolean} returns {headers, records, messages} or false
+   */
+  async function getOne({blobId, status}) {
+    const queue = generateQueueId({blobId, status});
+    try {
+      errorUndefinedQueue(queue);
+      const channelInfo = await channel.assertQueue(queue, {durable: true});
+      if (checkMessageCount(channelInfo)) {
+        debug(`Prepared to getOne from queue: ${queue}`);
+        // Returns false if 0 items in queue
+        const message = await channel.get(queue);
+
+        // debugData(`Message: ${inspect(message, {colors: true, maxArrayLength: 3, depth: 3})}`);
+        // Do not spam the logs
+
+        if (message) {
+          debugDev(`Got one from queue: ${queue}`);
+          const headers = getHeaderInfo(message);
+          const records = messagesToRecords([message]);
+          return {headers, records, messages: [message]};
+        }
+
+        return false;
+      }
+
+      return false;
+    } catch (error) {
+      handleAmqpErrors(error);
+    }
+  }
+
+  // MARK: Ack messages
+  /**
+   * Ack messages
```

Comment:
Comment could utilize non-shortened version of terms to clarify: ack -> acknowledge